### ISSUE TITLE
Nested rest destructuring does not work with parameters

### ISSUE THREAD

**Issue by sarimarton** [babel/babel#7860]
**Describe the bug**
Nested rest destructuring does not work with parameters.

**Input Code**
(function(...[x]) {})

```js
(function(...[x]) {})
```

### Babel Configuration (.babelrc, package.json, cli command)

Everything is maxed out/turned on/enabled etc.

**Expected behavior/code**
There is no syntax error thrown by Babel.

**Possible Solution**

**Environment**
- Babel version(s): [e.g. v6.0.0, v7.0.0-beta.34]
- Node/npm version: [e.g. Node 8/npm 5]
- OS: [e.g. OSX 10.13.4, Windows 10]
- Monorepo [e.g. yes/no/Lerna]
- How you are using Babel: cli

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by xtuc** [comment]
I can't reproduce the syntax error but I still don't have the correct behavior:

Babel 6 and 7:

```js
(function(...[x]) {x})
```

transpile to:

```js
(function () {
  x;
});
```

**Comment by sarimarton** [comment]
You're right @xtuc, it's my fault to have it detected as syntax error, and yeah, unfortunately the transformation still doesn't do the job. I have the same results as you.

**Comment by afhole** [comment]
We have also encountered this issue, e.g.
`const f = (...[a,b,...c]) => { console.log(a + b, c) };`

[babeljs.io/repl/](https://babeljs.io/repl/#?babili=true&browsers=&build=&builtIns=false&code_lz=MYewdgzgLgBAZjAvDAFAOgwbQIYBoBGuGawAugJRIB8MA3jKJCADYCmazIA5itjANQxCDSgF8A3EA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=false&presets=es2015%2Cbabili&prettier=false&targets=&version=6.26.0&envVersion=)
